core/pkg/plugin/plugin_interface.go

```diff
@@ -0,0 +1,55 @@
+package plugin
+
+import (
+	"encoding/gob"
+	"net/rpc"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/opencost/opencost/core/pkg/log"
+	"github.com/opencost/opencost/core/pkg/model"
+)
+
+// plugin interface
+type CustomCostSource interface {
+	GetCustomCosts(req model.CustomCostRequestInterface) []model.CustomCostResponse
+}
+
+// RPC impl
+type CustomCostRPC struct{ client *rpc.Client }
+
+func init() {
+	gob.Register(model.CustomCostRequest{})
+}
+func (c *CustomCostRPC) GetCustomCosts(req model.CustomCostRequestInterface) []model.CustomCostResponse {
+
+	var resp []model.CustomCostResponse
+	err := c.client.Call("Plugin.GetCustomCosts", &req, &resp)
+	if err != nil {
+		log.Errorf("error calling plugin: %v", err)
+	}
+
+	return resp
+}
+
+type CustomCostRPCServer struct {
+	// This is the real implementation
+	Impl CustomCostSource
+}
+
+func (s *CustomCostRPCServer) GetCustomCosts(args interface{}, resp *[]model.CustomCostResponse) error {
+	*resp = s.Impl.GetCustomCosts(args.(model.CustomCostRequestInterface))
+	return nil
+}
+
+type CustomCostPlugin struct {
+	// Impl Injection
+	Impl CustomCostSource
+}
+
+func (p *CustomCostPlugin) Server(*plugin.MuxBroker) (interface{}, error) {
+	return &CustomCostRPCServer{Impl: p.Impl}, nil
+}
+
+func (CustomCostPlugin) Client(b *plugin.MuxBroker, c *rpc.Client) (interface{}, error) {
```

Comment:
This signature is kind of funky; its on the `CustomCostPlugin` struct but doesn't use any of the fields?

Reply:
straight copy pasta from the go-plugin example. Here's what they have to say about it: 

> // Ignore MuxBroker. That is used to create more multiplexed streams on our
> // plugin connection and is a more advanced use case.
> func (GreeterPlugin) Client(b *plugin.MuxBroker, c *rpc.Client) (interface{}, error) {
> 	return &GreeterRPC{client: c}, nil
>  }

Could it be removed? Maybe. figured it was easier to just include it, and clean up if unused?

Reply:
Very odd! Would you link out to the source in a comment? After that I don't mind at all; I wonder how it works.

Reply:
commented with link 👍